app/services/prison_offender_manager_service.rb

```diff
@@ -3,16 +3,16 @@
 class PrisonOffenderManagerService
   # Note - get_poms_for and get_pom_at return different data...
   def self.get_poms_for(prison)
+    # This API call doesn't do what it says on the tin. It can return duplicate
+    # staff_ids in the situation where someone has more than one role.
     poms = Nomis::Elite2::PrisonOffenderManagerApi.list(prison)
-    pom_details = PomDetail.where(nomis_staff_id: poms.map(&:staff_id).map(&:to_i))
+    pom_details = PomDetail.where(nomis_staff_id: poms.map(&:staff_id))
 
-    poms = poms.map { |pom|
+    poms.map { |pom|
       detail = get_pom_detail(pom_details, pom.staff_id.to_i)
       pom.add_detail(detail, prison)
       pom
-    }.compact
-
-    poms
+    }.select { |pom| pom.prison_officer? || pom.probation_officer? }.uniq(&:staff_id)
```

Comment:
This might end up with people disappearing from the list of POMs where they might mistakenly have turned up.  Might be worth warning those on support of what the likely issue is if this occurs.

Reply:
Don't think so. We were crashing if the position came back as anything other that 'PO' or 'PRO' and also filtering later - so a POM was potentially there but not in either list when it came to allocation